pyupgrade.py

```diff
@@ -917,10 +917,16 @@ def _simplify_conversion_flag(flag: str) -> str:
     return ''.join(parts)
 
 
+NAMED_UNICODE_RE = re.compile(r'\\N{{([^}]+)}}')
```

Comment:
there's a slight issue with this:

```python
a = 1
print(r'\\N{snowman} %s' % (a,))
```

```console
$ python3 t.py.old
\\N{snowman} 1
```

```console
$ cp t.py.old t.py && pyupgrade t.py
Rewriting t.py
$ python3 t.py
Traceback (most recent call last):
  File "t.py", line 2, in <module>
    print(r'\\N{snowman} {}'.format(a))
KeyError: 'snowman'
```

I believe you need something like this: https://github.com/asottile-archive/textmate-highlight/commit/c52ec58b7cdde404baa0e53cf1b26a940bbcc4e4#diff-bcc683052c071bec26ad5c7ed6fc5d1fR44

Reply:
Thanks for the pointer, negative lookbehind (`(?<!...)`) was exactly what I needed.